File: sdk/node/neoroseVfrClient.js
// Minimal Node client for Neorose VFR
// Usage: const client = new VFRClient('http://127.0.0.1:8000');
//        const jobId = await client.tryOn('user.jpg','garment.jpg');
//        const status = await client.wait(jobId);

const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
const fs = require('fs');
const FormData = require('form-data');

class VFRClient {
  constructor(baseUrl) { this.baseUrl = baseUrl.replace(/\/$/, ''); }

  async tryOn(userPath, garmentPath, garmentSidePath) {
    const form = new FormData();
    form.append('user_image', fs.createReadStream(userPath));
    form.append('garment_front', fs.createReadStream(garmentPath));
    if (garmentSidePath) form.append('garment_side', fs.createReadStream(garmentSidePath));
    const r = await fetch(this.baseUrl + '/v1/jobs/tryon', { method: 'POST', body: form });
    if (!r.ok) throw new Error(await r.text());
    const js = await r.json();
    return js.job_id;
  }

  async status(jobId) {
    const r = await fetch(this.baseUrl + '/v1/jobs/' + jobId);
    if (!r.ok) throw new Error(await r.text());
    return r.json();
  }

  async wait(jobId, timeoutMs = 120000) {
    const start = Date.now();
    while (Date.now() - start < timeoutMs) {
      const s = await this.status(jobId);
      if (s.status === 'completed' || s.status === 'failed') return s;
      await new Promise(res => setTimeout(res, 1000));
    }
    return { job_id: jobId, status: 'timeout' };
  }

  async download(jobId, outPath) {
    const r = await fetch(this.baseUrl + '/v1/jobs/' + jobId + '/result');
    if (!r.ok) throw new Error(await r.text());
    const buf = await r.buffer();
    fs.writeFileSync(outPath, buf);
    return outPath;
  }
}

module.exports = { VFRClient };

File: sdk/python/neorose_vfr_client.py
import io
import os
import requests
from typing import Optional


class VFRClient:
    def __init__(self, base_url: str = "http://127.0.0.1:8000") -> None:
        self.base_url = base_url.rstrip("/")

    def try_on(
        self,
        user_image_path: str,
        garment_front_path: str,
        garment_side_path: Optional[str] = None,
    ) -> str:
        url = f"{self.base_url}/v1/jobs/tryon"
        files = {
            "user_image": (os.path.basename(user_image_path), open(user_image_path, "rb"), "image/jpeg"),
            "garment_front": (os.path.basename(garment_front_path), open(garment_front_path, "rb"), "image/jpeg"),
        }
        if garment_side_path:
            files["garment_side"] = (
                os.path.basename(garment_side_path),
                open(garment_side_path, "rb"),
                "image/jpeg",
            )
        r = requests.post(url, files=files, timeout=60)
        r.raise_for_status()
        return r.json()["job_id"]

    def job_status(self, job_id: str) -> dict:
        r = requests.get(f"{self.base_url}/v1/jobs/{job_id}", timeout=30)
        r.raise_for_status()
        return r.json()

    def download_result(self, job_id: str) -> bytes:
        r = requests.get(f"{self.base_url}/v1/jobs/{job_id}/result", timeout=60)
        r.raise_for_status()
        return r.content

